### PR TITLE
feat: create snapshot reference image subfolders for iOS 14 and 15 simulators SQPIT-808

### PR DESCRIPTION
**Pull request by billypchan**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQPIT-808" title="SQPIT-808" target="_blank">TaskSQPIT-808</a>  snapshot folder for iOS simulator 14 and 15
  </td></table>
  <br />
 

# What's new in this PR?

### Issues

The coming release will be built with Xcode 12 and test with iOS 14 simulator. The current snapshot reference images are created with iOS 15 simulator and the compare results are mismatch.

### Causes 

We have only one set of reference image for all simulator version

### Solutions

Create subfolder under `SNAPSHOT_REFERENCE_DIR` env with iOS version. e.g. `wire-ios/Wire-iOS Tests/ReferenceImages/15_0_0/`. Move the reference images to that folder. And copy the reference images from last release branch to `14_0_0` subfolder.

### CODE REVIEW

File: Wire-iOS Tests/SwiftSnapshotTesting/XCTestCase+SnapshotTesting.swift [hunk 1]
```diff
@@ -267,9 +267,18 @@ extension XCTestCase {
...
+
+        return "\(os.majorVersion)_\(os.minorVersion)_\(os.patchVersion)"
+    }
+
     func snapshotDirectory(file: StaticString = #file) -> String {
         let fileName = "\(file)"
-        let path = ProcessInfo.processInfo.environment["SNAPSHOT_REFERENCE_DIR"]! + "/" + URL(fileURLWithPath: fileName).deletingPathExtension().lastPathComponent
+        let path = ProcessInfo.processInfo.environment["SNAPSHOT_REFERENCE_DIR"]! + "/" +
+        osVersionString + "/" +
+        URL(fileURLWithPath: fileName).deletingPathExtension().lastPathComponent
```
Comment (johnxnguyen) [inline Wire-iOS Tests/SwiftSnapshotTesting/XCTestCase+SnapshotTesting.swift]: Does this work for both Swift and ObjC snapshot frameworks?

**Comment by johnxnguyen** [comment]
I also don't see the iOS 14 snapshots, did you forget to add them?

File: Wire-iOS Tests/SwiftSnapshotTesting/XCTestCase+SnapshotTesting.swift (hunk 1 above)
Comment (billypchan) [reply Wire-iOS Tests/SwiftSnapshotTesting/XCTestCase+SnapshotTesting.swift]: only swift snapshot testing result is affected. FB snapshots usually not affected by the simulator version. (It is a known issue)

**Comment by billypchan** [comment]
@johnxnguyen I got a framework issue when building with Xcode 12. Will open a new PR for adding Xcode 12 snapshots.

File: Wire-iOS Tests/SwiftSnapshotTesting/XCTestCase+SnapshotTesting.swift (hunk 1 above)
Comment (johnxnguyen) [reply Wire-iOS Tests/SwiftSnapshotTesting/XCTestCase+SnapshotTesting.swift]: Good to know.

**Review by johnxnguyen — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**